platform/ext/target/nordic_nrf/Native_Driver/spu.c

```diff
@@ -0,0 +1,164 @@
+
+/* Minimum size of Non-Secure Callable regions. */
+#define FLASH_NSC_MIN_SIZE 32
+
+/* Maximum size of Non-Secure Callable regions. */
+#define FLASH_NSC_MAX_SIZE 4096
+
+
+/* Convenience macros for SPU attributes */
+#define FLASH_EXEC                                                             \
+	((SPU_FLASHREGION_PERM_EXECUTE_Enable                                  \
+	  << SPU_FLASHREGION_PERM_EXECUTE_Pos) &                               \
+	 SPU_FLASHREGION_PERM_EXECUTE_Msk)
+
+#define FLASH_WRITE                                                            \
+	((SPU_FLASHREGION_PERM_WRITE_Enable                                    \
+	  << SPU_FLASHREGION_PERM_WRITE_Pos) &                                 \
+	 SPU_FLASHREGION_PERM_WRITE_Msk)
+
+#define FLASH_READ                                                             \
+	((SPU_FLASHREGION_PERM_READ_Enable << SPU_FLASHREGION_PERM_READ_Pos) & \
+	 SPU_FLASHREGION_PERM_READ_Msk)
+
+#define FLASH_SECURE                                                           \
+	((SPU_FLASHREGION_PERM_SECATTR_Secure                                  \
+	  << SPU_FLASHREGION_PERM_SECATTR_Pos) &                               \
+	 SPU_FLASHREGION_PERM_SECATTR_Msk)
+
+#define FLASH_NONSEC                                                           \
+	((SPU_FLASHREGION_PERM_SECATTR_Non_Secure                              \
+	  << SPU_FLASHREGION_PERM_SECATTR_Pos) &                               \
+	 SPU_FLASHREGION_PERM_SECATTR_Msk)
+
+#define SRAM_EXEC                                                              \
+	((SPU_RAMREGION_PERM_EXECUTE_Enable                                    \
+	  << SPU_RAMREGION_PERM_EXECUTE_Pos) &                                 \
+	 SPU_RAMREGION_PERM_EXECUTE_Msk)
+
+#define SRAM_WRITE                                                             \
+	((SPU_RAMREGION_PERM_WRITE_Enable << SPU_RAMREGION_PERM_WRITE_Pos) &   \
+	 SPU_RAMREGION_PERM_WRITE_Msk)
+
+#define SRAM_READ                                                              \
+	((SPU_RAMREGION_PERM_READ_Enable << SPU_RAMREGION_PERM_READ_Pos) &     \
+	 SPU_RAMREGION_PERM_READ_Msk)
+
+#define SRAM_SECURE                                                            \
+	((SPU_RAMREGION_PERM_SECATTR_Secure                                    \
+	  << SPU_RAMREGION_PERM_SECATTR_Pos) &                                 \
+	 SPU_RAMREGION_PERM_SECATTR_Msk)
+
+#define SRAM_NONSEC                                                            \
+	((SPU_RAMREGION_PERM_SECATTR_Non_Secure                                \
+	  << SPU_RAMREGION_PERM_SECATTR_Pos) &                                 \
+	 SPU_RAMREGION_PERM_SECATTR_Msk)
+
+#define FLASH_RWX (FLASH_READ | FLASH_WRITE | FLASH_EXEC)
+#define SRAM_RWX (SRAM_READ | SRAM_WRITE | SRAM_EXEC)
+
+
+#define FLASH_NSC_REGION(reg_number)                                           \
+	(((reg_number) << SPU_FLASHNSC_REGION_REGION_Pos) &                    \
+	SPU_FLASHNSC_REGION_REGION_Msk)
```

Comment:
SPU HAL is responsible for doing the proper shifting and masking, so such macro should not be used in code that calls `nrf_spu_flashnsc_set()`. Same goes for `FLASH_NSC_SIZE`.

Reply:
Agreed, that should be fixed.
@oyvindronningstad ^^

Reply:
@anangl btw,
```
    p_reg->FLASHNSC[flash_nsc_id].REGION = (uint32_t)region_number |
        (lock_conf ? SPU_FLASHNSC_REGION_LOCK_Msk : 0);
    p_reg->FLASHNSC[flash_nsc_id].SIZE = (uint32_t)flash_nsc_size |
        (lock_conf ? SPU_FLASHNSC_SIZE_LOCK_Msk : 0);
```

there is no shift, mask applied on region and size, but i guess this is OK if shift should be 0. But strictly speaking, the shifting here is (was) not redundant.

Reply:
Fixed now.